src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs

```diff
@@ -54,8 +55,8 @@ private static bool GetOboOrAppKey(AuthenticationRequestParameters requestParame
 
             if (requestParameters.ApiId == TelemetryCore.Internal.Events.ApiEvent.ApiIds.AcquireTokenForClient)
             {
-                key = requestParameters.AppConfig.ClientId + "_AppTokenCache";
-                return true;
+                string tenantId = requestParameters.Authority.TenantId ?? "";
```

Comment:
you're not using the tenantId var in your returned result

Reply:
also if tenantId is an empty string, key will have double underscores, i.e. clientId__AppTokenCache

Reply:
This is fine, I think it's more readable this way, to show that a tenantId does not exist (i.e. ADFS authority)